src/main/java/salesforce/ui/pages/BasePage.java

```diff
@@ -1,25 +1,26 @@
 package salesforce.ui.pages;
 
+import core.WebDriverManagerSingleton;
+import core.WebElementAction;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.support.PageFactory;
 import org.openqa.selenium.support.ui.WebDriverWait;
-import utils.SetUp;
 
 /**
  * This abstract class is the base page for all pages.
  */
 public abstract class BasePage {
     protected WebDriver driver;
     protected WebDriverWait wait;
+    protected WebElementAction webElementAction;
 
     /**
      * Constructor method.
-     *
-     * @param driver with the driver needed.
      */
-    public BasePage(final WebDriver driver) {
-        this.driver = driver;
-        wait = new WebDriverWait(driver, Long.valueOf(SetUp.WAIT_TIME_VALUE.getValue()));
+    public BasePage() {
```

Comment:
If a driver is instanciated here, everytime a new page is clicked a new driver will bw instanciated and open a new browser

Reply:
Thanks, I have fixed that